spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/BucketizedCol.scala

```
@@ -0,0 +1,85 @@
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.{DenseTensorApply, DoubleType, FloatType, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import java.util.Arrays.binarySearch
+
+import scala.reflect.ClassTag
+
+/**
+ * BucketizedCol operation represents discretized dense input.
+ *
+ * The Operation can handle single or multi feature column,
+ * as long as the boundaries is same between feature columns.
+ *
+ * Buckets include the left boundary, and exclude the right boundary.
+ * Namely, boundaries=Array(0, 1, 10) generates buckets (-inf,0),[0,1),[1,10),[10,+inf)
+ *
+ * For example, boundaries = Array(0, 10, 100) and input tensor is an 2D 3x2 DenseTensor:
+ *  -1, 1
+ *  101, 10
+ *  5, 100
+ *
+ *  the output tensor should be an 2D 3x2 DenseTensor
+ *  0, 1
+ *  3, 2
+ *  1, 3
+ *
+ * @param boundaries The bound Array of each bucket.
+ * @tparam T Numeric type. Parameter tensor numeric type. Only support float/double now
+ */
+
+class BucketizedCol[T: ClassTag](
+  boundaries: Array[Double])(implicit ev: TensorNumeric[T])
+  extends Operation[Tensor[T], Tensor[Int], T] {
+
+  private val boundariesImpl = boundaries.map(ev.fromType[Double])
+  output = Tensor[Int]()
+
+  override def updateOutput(input: Tensor[T]): Tensor[Int] = {
+
+    val rows = input.size(dim = 1)
+    val cols = input.size(dim = 2)
+    val resTensor = Tensor[Int](rows, cols)
+    require(boundaries.length >= 1,
+    "the length of boundaries must be more than or equal to 1")
+
+    ev.getType() match {
+      case FloatType =>
+        resTensor.applyFun[Float](
+          input.asInstanceOf[Tensor[Float]],
+          x => math.abs(binarySearch(boundariesImpl.asInstanceOf[Array[Float]], x) + 1))
+      case DoubleType =>
+        resTensor.applyFun[Double](
+          input.asInstanceOf[Tensor[Double]],
+          x => math.abs(binarySearch(boundariesImpl.asInstanceOf[Array[Double]], x) + 1))
```

should we throw an exception for not support numeric type here?